common/main.c

```diff
@@ -5,19 +5,36 @@
 
 #include <compiler.h>
 #include <console.h>
+#include <debug.h>
 #include <dm.h>
 #include <stdbool.h>
 #include <work.h>
+#include <drivers/watchdog.h>
 #include <platform/devices.h>
 
+#define WDOG_TIMEOUT (5 * 1000 * 1000) /* 5 seconds */
+
 noreturn void main(void);
 
+void main(void);
+
 noreturn void
 main(void)
 {
+	struct device *twd_dev;
```

Comment:
We don't care if this is a `twd` or some other kind of watchdog. We just ask for any watchdog device with `dm_get_by_class`, so this should be named something like `watchdog` (the `_dev` is also redundant).

Reply:
Done.